cellxgene_resources/curation_qa.ipynb

```diff
@@ -656,6 +657,129 @@
+    "    '''\n",
+    "    Input: mean expression of gene\n",
+    "    Output: assignment of sex\n",
+    "    '''\n",
+    "    if x >= 0.5:\n",
+    "        return 'Male'\n",
+    "    elif x < 0.2:\n",
+    "        return 'Female'\n",
+    "    else:\n",
+    "        pass\n",
+    "\n",
+    "def calculate_sex(region_dict):\n",
+    "    '''\n",
+    "    Input: dictionary containing dataframes for PAR and Non-PAR average expression counts per gene per donor\n",
+    "    Output: a dataframe of both nonpar and par summed expression counts per donor with sex assignment\n",
+    "    '''\n",
+    "    par_df = region_dict['par'][1]\n",
+    "    par_df['par_sum'] = par_df.sum(numeric_only=True, axis=1)\n",
+    "    nonpar_df = region_dict['nonpar'][1]\n",
+    "    nonpar_df['nonpar_sum'] = nonpar_df.sum(numeric_only=True, axis=1)\n",
+    "    nonpar_par_df = pd.merge(nonpar_df, par_df, how='left', left_index=True, right_index=True) \n",
+    "    nonpar_par_df['nonPAR_to_PAR'] = nonpar_par_df['nonpar_sum']/nonpar_par_df['par_sum']\n",
+    "    nonpar_par_df.rename(columns={'donor_id_x':'donor_id'}, inplace=True)\n",
+    "    nonpar_par_df['scRNAseq_sex'] = nonpar_par_df['nonPAR_to_PAR'].apply(assign_sex)\n",
+    "    return nonpar_par_df\n",
+    "\n",
+    "def check_donor_sex(nonpar_par_df, adata):\n",
+    "    '''\n",
+    "    Input: nonpar_par dataframe and original anndata object\n",
+    "    Output: print out of only the inconsistent donor's nonPAR_to_par ratio, scRNAseq-calculated sex, and author-labeled sex\n",
+    "    '''\n",
+    "    donors = adata.obs.value_counts(['donor_id', 'sex']).reset_index().rename(columns={\"index\": \"value\", 0: \"count\"})\n",
+    "    donors.drop(columns=['count'],inplace=True)\n",
+    "    check_donors = pd.merge(nonpar_par_df, donors, how='left', left_on='donor_id', right_on='donor_id')\n",
+    "    check_donors['mismatched'] = check_donors['scRNAseq_sex'].str.lower() != check_donors['sex'].str.lower()\n",
+    "    mismatches = check_donors[check_donors['mismatched'] == True]\n",
+    "    print(f\"Inconsistent Donor(s): \\n{mismatches[['donor_id', 'nonPAR_to_PAR', 'scRNAseq_sex', 'sex']]}\\n\")\n",
+    "    \n",
+    "def check_percent(par_adata,nonpar_adata,par_ids,nonpar_ids):\n",
+    "    '''\n",
+    "    Input: subset adatas and id lists\n",
+    "    Output: percent of genes found per region\n",
+    "    '''\n",
+    "    print(f\"% PAR genes found: {(par_adata.shape[1]/len(par_ids['gene_ids']))*100}\")\n",
+    "    print(f\"% Non-PAR genes found: {(nonpar_adata.shape[1]/len(nonpar_ids['gene_ids']))*100}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Run scRNA-seq sex analysis**<br>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Get PAR & Non-PAR gene_ids from ref_file\n",
+    "ref_file = 'ref_files/gtf.tsv'\n",
```

Comment:
This file does not exist
And I would choose a more descriptive variables than `ref`/`ref_file`

Reply:
Sample suggestion to simplify things...
`chry_file = 'ref_files/chry_regions.json'`
`chry_genes = json.load(open(chry_file))`
`par_ids = chry_genes['PAR']`
`nonpar_ids = chry_genes['NONPAR']`